src/StepManager.cpp

```diff
@@ -114,12 +115,34 @@ MatchResult StepManager::stepMatches(const std::string &stepDescription) {
         const boost::shared_ptr<const StepInfo>& stepInfo = iter->second;
         SingleStepMatch currentMatch = stepInfo->matches(stepDescription);
         if (currentMatch) {
+            for (auto& submatch : currentMatch.submatches) {
+                submatch.position = codePointsFromStart(stepDescription, submatch.position);
+            }
             matchResult.addMatch(currentMatch);
         }
     }
     return matchResult;
 }
 
+
+std::ptrdiff_t StepManager::codePointsFromStart(const std::string &expression, size_t index) {
+    std::ptrdiff_t codePointCount(0);
+    for (const uint8_t i : expression) {
```

Comment:
@src-ableton `cucumber-cpp` still supports C++98, you can't use for-range loops.